kroxylicious-kms-provider-hashicorp-vault/src/main/java/io/kroxylicious/kms/provider/hashicorp/vault/config/JdkTls.java

```diff
@@ -0,0 +1,134 @@
+        }
+
+        // suppressing sonar security warning as we are intentionally throwing security out the window
+        @SuppressWarnings("java:S4830")
+        @Override
+        public void checkServerTrusted(X509Certificate[] chain, String authType) {
+            // do nothing, the api is to throw if not trusted
+        }
+
+        @Override
+        public X509Certificate[] getAcceptedIssuers() {
+            return new X509Certificate[0];
+        }
+    };
+    public static final TrustManager[] INSECURE_TRUST_MANAGERS = { INSECURE_TRUST_MANAGER };
+
+    public SSLContext sslContext() {
+        try {
+            if (tls != null && tls.trust() != null) {
+                return customSSLContext(tls.trust());
+            }
+            else {
+                return SSLContext.getDefault();
+            }
+        }
+        catch (Exception e) {
+            throw new SslConfigurationException(e);
+        }
+    }
+
+    @NonNull
+    private SSLContext customSSLContext(TrustProvider trust) throws NoSuchAlgorithmException, KeyManagementException {
+        Optional<TrustManager[]> trustManagers = getTrustManagers(trust);
+        return trustManagers.map(JdkTls::getSslContext).orElse(SSLContext.getDefault());
+    }
+
+    @NonNull
+    private static SSLContext getSslContext(TrustManager[] trustManagers1) {
+        try {
+            SSLContext rsa = SSLContext.getInstance("TLS");
+            rsa.init(null, trustManagers1, new SecureRandom());
+            return rsa;
+        }
+        catch (Exception e) {
+            throw new SslConfigurationException(e);
+        }
+    }
+
+    /* exposed for testing */
+    static Optional<TrustManager[]> getTrustManagers(TrustProvider trust) {
+        return trust.accept(new TrustProviderVisitor<>() {
+            @Override
+            public Optional<TrustManager[]> visit(TrustStore trustStore) {
+                try {
+                    KeyStore instance = KeyStore.getInstance(trustStore.getType());
+                    char[] charArray = trustStore.storePasswordProvider() != null ? trustStore.storePasswordProvider().getProvidedPassword().toCharArray() : null;
+                    instance.load(new FileInputStream(trustStore.storeFile()), charArray);
+                    TrustManagerFactory managerFactory = TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm());
+                    managerFactory.init(instance);
+                    return Optional.of(managerFactory.getTrustManagers());
+                }
+                catch (Exception e) {
+                    throw new SslConfigurationException(e);
+                }
+            }
+
+            @Override
+            public Optional<TrustManager[]> visit(InsecureTls insecureTls) {
+                if (insecureTls.insecure()) {
+                    return Optional.of(INSECURE_TRUST_MANAGERS);
+                }
+                else {
+                    return Optional.empty();
```

Comment:
Maybe just returning the platform's default trust managers would be consistent.   I think that means the need for the Optional goes..



```suggestion
                        return Optional.of(TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm()).getTrustManagers());
```

Reply:
have updated to do this, you also have to init the TrustManagerFactory with a `null` KeyContext which eventually returns a TrustManager that trusts default certs like `trustManager = getInstance(TrustStoreManager.getTrustedCerts());`. It's not really documented that that should be the behaviour for null though. We have IT and unit tests showing it rejects the server self-signed-cert, but can we depend on this across platforms?

```
    /**
     * Initializes this factory with a source of certificate
     * authorities and related trust material.
     * <P>
     * The provider typically uses a KeyStore as a basis for making
     * trust decisions.
     * <P>
     * For more flexible initialization, please see
     * {@link #init(ManagerFactoryParameters)}.
     *
     * @param ks the key store, or null
     * @throws KeyStoreException if this operation fails
     */
    public final void init(KeyStore ks) throws KeyStoreException {
        factorySpi.engineInit(ks);
    }
    ```